### ISSUE TITLE
Suggestion: Village Progression Seems Steep

### ISSUE THREAD

**Issue by JohnyDL** [TangoTek/TekTopia-Community#377]
**Describe the Feature**
Rebalancing the economy/villager skills a little so that the village feels like it's progressing right from the start and encourages players to keep playing.

**Why do you think this improvement will help the mod?**
Combining several features makes the Economy a bit... Challenging... Especially for more casual players. Even in creative I'm finding it a struggle to have fun playing in my restart. Not everyone is super hardcore and there's a combination of things that makes playing this a grind from the first second.
- Token cost increases by 20% for every 5 tokens brought
- Merchant devalues things you make lots of overtime to the point they don't show up
- Children Villagers are stuck with 0 intelligence until you have a school
- Villagers from captured Villages are stuck with 0 intelligence until you have a library
- All Nomad converts are stuck with no progress until you have a library
- Villagers return products based on their job skill level in most cases so it's low when at very low int/job levels.

The suggested changes below should not eliminate all challenges and any one of them might improve the early-game playability and allow players to feel progression and encourage them to keep playing more and more. I've also gone to extra effort to keep (or even amplify) those mid- to late-game challenges for those that want/need them without making things trivial.

**Additional context**
Merchant:
- Change the size of his history to scale with the number of villagers the village has. The bigger the village is the more memorable it is the more the history of transactions should change the merchant's economy. A very small village is then not hit by the massive price swings and lockouts while a very big village still has the Merchant economy challenge, it might even be harder
- Remove history over time. The longer ago a trade was made the less it should impact the merchant's economy. I propose simply deleting old history, again this could be scaled so maybe for with a village of n population it persists for n+9 minecraft days in single player and 3n+72 minecraft days on servers, assuming that no AFK that's 3 hours of play for the trades to be reset on single player to start plus an additional 20 minutes for every villager while on the servers since the clock ticks when players aren't online it's 1 day plus 1 hour for every villager. This erasure time is decided when the history is made, if your village grows in the meanwhile then that doesn't make the history stay longer than it would have.
- Make the economy slightly positive sum the more you sell the more demand you create for your goods, with the scaled and time-dependent history there is room for more progression and more variation as the village gets bigger, it also relies on active play and punishes AFK/waiting to get the best out of the economy while small villages are unaffected.

Architect and Tradesman:
- Don't base the price off the number of tokens brought. It severely punishes villager job changes and when a village takes a blow like with villagers dying to starvation or mobs it can feel almost irrecoverable. If someone puts in 10 hours of effort to earn their guards and then the price of replacing them goes up 20% then their guards dying costs 12 further hours of play for no progression. It might even be that the guards dying costs the village even more in the long term too, more dying, and longer-term price hikes, a handful of early game deaths can effectively add hours of work to every stage of the game going forward. Token prices should, therefore, be entirely based on the state of the village, not the number of tokens purchased.
- First Distinguish between "one-off" tokens and "repeat" tokens. Generally speaking, you only need 1 butcher building, 1 of each animal pen, 1 blacksmith, 1 library, 1 school, 1 tavern, 1 kitchen, while bedrooms, mines, guard posts and the profession tokens are multi-purchase.
- One-off tokens should scale with the size/state of the village as a whole. The decision which of these to buy should maybe as much as double the cost of the next one of these upgrades. Each and every one of these should be a hard decision and only get harder with each that you buy. Exponentiating the price also means the upfront cost of the tavern/school/library can be reduced a little. Cows are also the superior animal being meat and leather and milk, so maybe their food quality needs to be nerfed relative to pig and sheep to make the decision harder.
- Bedrooms I'm not sure how exactly these should scale, maybe based on the number of villagers, so each bed costs population/10 and then each bedroom size has some base cost too. This would mean for players balancing the cost of buying houses before you need them with emeralds later since buying them early means they're cheaper but it means pushing off other progression. Their cost also needs to be balanced with the ability to loot natural villages for bedroom tokens (see #165) So I don't think this can be an effective emerald sink
- Mines and Guard Posts should scale directly on how many of these structures have been brought and should maybe scale quadratically to provide a nice emerald sink for the economy as changing where the mines is can be annoying and also stops the Druids refilling them, while guards work best if there's 1 post per guard and you might need several guards for a necro attack. (might need ways of further compressing emeralds than emerald blocks as this gets bigger)
- Profession Tokens this is a little more complicated since it involves tracking the use of the tokens but my suggestion would be to scale the price based off the number of that type of villager + the number of unused tokens the player has brought. This is relatively self-balancing, your first farmer should always cost the same no matter whether you just made the village, or whether the village was just wiped out and you're restarting. Converting a villager reduces the cost to convert them back. Over time the cost of villagers increases as you have more of them and maybe this doesn't have to be linear. Each villager's cost can also be scaled independently, if you expect a village to need 1 blacksmith for every 10 farmers, miners, lumberjacks and guards then their prices increase 10x faster.

Children
- could be born with the average intelligence/10 and a little variance. The smarter your village is the smarter it's children naturally turn out to be. Early game the variance plays a big role if the variance is +-5 levels then a child might be half as good as even the best villagers out of the womb, while late game with an average intelligence of 90 for the village the children don't get that much of an extra kick start and schooling is very important
- children could have a 1/100 chance of getting an intelligence level up when they see an adult skill up (even if that adult skilling up attempt is clamped by their own intelligence) nowhere near as powerful as a school but children do learn by watching.

Adults
- The intelligence clamping could be a little more lax using one of these functions
1. int + x = Max skill 
- They can get a few more skills than their intelligence
2. Max(x,int) = Max skill
- They can get a few job skills to start but they will have an intelligence wall to overcome eventually
3. floor(100-(100-int)*(100-x)/100) = Max skill
- They have a minimum skill level but they never have an intelligence wall and still need to get to 100 intelligence to get 100 job levels
4. Something else that provides a steeper learning curve later but allows for nearly linear progression and a boost near 0 intelligence?
- Adults without a library have a 1/(job level)^6 chance of getting an intelligence boost from working at their max job skill and attempting to level up that job. If their job level is 10 then they have a 1/1,000,000 chance every time they skill up but if they're job level 99 then they only have a 1/941,480,149,401 chance allowing for some skill progression in absence of the library even if it's a one in a million shot, the vast majority of the time this would be a lot less than a library but it allows some progression especially at those crucial low levels that hurt so critically to be without, a villager with 0 or 1 intelligence always gets a level up but to climb from 2 to 3 it's 1/32, and 3 to 4 is 1/729 it quickly tails off but there's still a chance and the chances soon add up.

**Comment by orangense** [comment]
I agree with this. I think size of the village is a great indicator of how "challenging" things should be allowed to become. That way there's no huge advantage of starting with a vanilla village. I am finding it very difficult as a casual player in my startup village to advance at all. It seems that I'm stuck because the items I can sell to the merchant are giving me less and less emeralds and the cost of the building tokens keep getting higher and higher. Purchasing more tokens from the tradesman to try to speed up the production of my farms and lumberjacks has only increased the cost of the architect tokens making it even harder to reach those goals.

**Comment by TangoTek** [comment]
Deciding what to base the price of structures was indeed a very hard decision that I went back and forth on quite a bit.  However, in the end this is the simple situation I'm trying to avoid.

1) Take down town hall  2) Walk 300 blocks 3) Place town hall token on a 6 block cave in a mountain 4) Buy tons of super cheap tokens from merchant that shows up because your village has no villagers, no buildings, and no history.  5) return to normal town and replace TH token

**Comment by JohnyDL** [comment]
I thought you persisted that stuff in the Town hall marker's NBT
I also thought that you wanted to make it so structure markers/profession markers weren't mix and match
Personally, I want every village to be it's own thing (maybe some NBT data) so you can only sell items made in a village in that village. Villagers can use village items from another village, and they get treated like village items so the bonuses on durability which allows player to player trade, if they craft something out of entirely one village's items it keeps that village's tag, if there's a mix then it becomes vanilla.
Vanilla emeralds or "unmarked" tokens (from naturally spawning locations) can be used anywhere, but a token sold from a village or purchased in a village only works in that village.

When a town hall is placed it gets a random 32bit ID & Location
When town hall is moved if moved greater than 120 blocks new random ID
When town hall is moved less than 120 blocks set countdown to update location. (if it takes 36 hours of loading the token to move it 300 blocks and back without breaking the ID and a player wants to spend hours doing that to get cheap tokens rather than playing the game and paying the prices all the power to them)
When a village item is created in the village it gets given that ID
When crafting if all IDs match result gets ID
When crafting if one ID fails to match get Vanilla
When "unmarked" token used in Village assign ID to it.
Trying to use a village profession token with the wrong ID will have no effect.
Using the /village item command has to be done inside a village and will tag it with that village's ID
A Village item placed in the storage of another village will turn red, rather than green, to indicate it can't be sold here

This way moving the village over a certain distance makes all the old things you collected relatively worthless in the "new" village but if it keeps all the data about the old village so the costs of tokens will remain the same, and now you have to rebuy half that stuff. Moving it back doesn't necessarily undo that mistake, you tried to cheat the system and so the game punished you for it by making your things worthless and your old tokens not work anymore, you're best of making a new town hall marker and starting your village structure token process from scratch.

There might be a command that could erase the location so if a player wants to move their village on the server that can be done while keeping all the IDs the same. And of course players could use that in Single player mode but the only people they're cheating then are themselves
